File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/dispatchlog/DispatchlogService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.dispatchlog;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class DispatchlogService extends BaseService {

	public DispatchlogService(Session session) {
		super(session);
	}

	public void deleteDispatchLog(long dispatchLogId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("dispatchLogId", dispatchLogId);

			_command.put("/dispatch.dispatchlog/delete-dispatch-log", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject getDispatchLog(long dispatchLogId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("dispatchLogId", dispatchLogId);

			_command.put("/dispatch.dispatchlog/get-dispatch-log", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getDispatchLogs(long dispatchTriggerId, int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("dispatchTriggerId", dispatchTriggerId);
			_params.put("end", end);
			_params.put("start", start);

			_command.put("/dispatch.dispatchlog/get-dispatch-logs", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getDispatchLogsCount(long dispatchTriggerId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("dispatchTriggerId", dispatchTriggerId);

			_command.put("/dispatch.dispatchlog/get-dispatch-logs-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

}

File: windows/Liferay.SDK/Service/V73/Dispatchlog/DispatchlogService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Dispatchlog
{
	public class DispatchlogService : ServiceBase
	{
		public DispatchlogService(ISession session)
			: base(session)
		{
		}

		public async Task<IEnumerable<dynamic>> GetDispatchLogsAsync(long dispatchTriggerId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("dispatchTriggerId", dispatchTriggerId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/dispatch.dispatchlog/get-dispatch-logs", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetDispatchLogsCountAsync(long dispatchTriggerId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("dispatchTriggerId", dispatchTriggerId);

			var _command = new JsonObject()
			{
				{ "/dispatch.dispatchlog/get-dispatch-logs-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task DeleteDispatchLogAsync(long dispatchLogId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("dispatchLogId", dispatchLogId);

			var _command = new JsonObject()
			{
				{ "/dispatch.dispatchlog/delete-dispatch-log", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> GetDispatchLogAsync(long dispatchLogId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("dispatchLogId", dispatchLogId);

			var _command = new JsonObject()
			{
				{ "/dispatch.dispatchlog/get-dispatch-log", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}